infer/lib/audio.py

```diff
@@ -39,19 +40,102 @@ def load_audio(file: str, sr: int) -> np.ndarray:
         raise FileNotFoundError(f"File not found: {file}")
 
     try:
-        # https://github.com/openai/whisper/blob/main/whisper/audio.py#L26
-        # This launches a subprocess to decode audio while down-mixing and resampling as necessary.
-        # Requires the ffmpeg CLI and `ffmpeg-python` package to be installed.
-        file = str(clean_path(file))  # 防止小白拷路径头尾带了空格和"和回车
-        out, _ = (
-            ffmpeg.input(file, threads=0)
-            .output("-", format="f32le", acodec="pcm_f32le", ac=1, ar=sr)
-            .run(cmd=["ffmpeg", "-nostdin"], capture_stdout=True, capture_stderr=True)
-        )
+        container = av.open(file)
+        resampler = AudioResampler(format="fltp", layout="mono", rate=sr)
+
+        # AV stores duration in nanoseconds
+        decoded_audio = (
+            ((container.duration * sr / container.bit_rate) // 1_000_000) + 1
+        ) * []
+
+        for frame in container.decode(audio=0):
+            frame.pts = None  # Clear presentation timestamp to avoid resampling issues
+            resampled = resampler.resample(frame)
+            decoded_audio.append(np.array(resampled))

```

Comment:
So the append can be a direct copy to that array, like
```py
decoded_audio[offset:offset+this_frame_len]
```

Reply:
> So the append can be a direct copy to that array, like
> 
> ```python
> decoded_audio[offset:offset+this_frame_len]
> ```

So... we need to cache the offset after each call for this, right?

Reply:
Implemented by 953b2d6